nautobot/extras/health_checks.py

```diff
@@ -26,3 +27,45 @@ def check_status(self):
             raise ServiceReturnedUnexpectedResult("Integrity Error")
         except DatabaseError:
             raise ServiceUnavailable("Database error")
+
+
+class RedisBackend(BaseHealthCheckBackend):
+    """Health check for Redis."""
+
+    redis_url = getattr(settings, "CACHEOPS_REDIS", "redis://localhost:6379/1")
+    sentinel_url = getattr(settings, "CACHEOPS_SENTINEL", None)
+
+    def check_status(self):
+        """Check Redis service by pinging the redis instance with a redis connection."""
+        # if CACHEOPS_REDIS is set to False or Sentinel is enabled
+        # We need to check Redis using Sentinel
+        if not self.redis_url or self.sentinel_url is not None:
+
+            from redis.sentinel import Sentinel
+
+            try:
+                sentinel = Sentinel(self.sentinel_url["locations"], socket_timeout=self.sentinel_url["socket_timeout"])
+                with sentinel.master_for(self.sentinel_url["service_name"], db=self.sentinel_url["db"]) as master:
+                    master.ping()
+            except ConnectionRefusedError as e:
```

Comment:
Most of these exceptions result in the same error message minus the execution type name. This could be cleaned up by combining those into a single except block and treating the type name as a variable in the string.

Something like this:

```python
except (ConnectionRefusedError, exceptions.ConnectionError, exceptions.TimeoutError) as e:
    self.add_error(ServiceUnavailable("Unable to connect to Redis Sentinel: %s" % type(e).__name__), e)
```

Reply:
I'll change it tomorrow